### DISCUSSION TITLE
What is the fundamental data structure?

### DISCUSSION THREAD

**Discussion by DavidLandes** [el-apps/Home-Maintenance#2] [Q&A]
How can we group the equipment & group lists below? The equipment will have a `manufacturer`, `part number`, `part list`, & `maintenance tasks`. These properties do not apply to `Home` or `Yard`. Should there be entirely different data types or should we find a common structure for everything in the lists?

### Equipment 
- Cars/Trucks
    - Oil Change
    - Tire Rotation
    - Light Replacement
    - Serpentine Belt
    - Car Wash
    - Windshield Wiper Replacement
- Lawn Mower
    - Oil Change
    - Belt Replacement
    - Blade Sharpening
    - Fuel Filter Replacement
    - Pully Replacement
    - Grease Pullies
- Heat Pump
    - Clean Radiator
- Furnace
    - Replace Filter
- Water Softener
    - Refill Softener Salt
    - Refill Cleaning Resin
- Dryer Vent
    - Clean Lint

### Groups
- File Storage
    - Perform backup
- Password manager
    - Rotate passwords
-  Home
    - Deep clean
        - It'll depend on the family, but this would be a year or quarterly kind of thing 
    - Inventory audit
        - If I had an inventory app, I would want to do an annual or bi-annual check of everything in it to see what we can throw out 
- Crawlspace
    - Basic check
        - I should probably do this at least quarterly, just to check for leaks, rust, etc.
- Yard
    - Mow
    - Weed eat
    - Mulch

**Comment by DavidLandes** [comment]
My original idea was to couple equipment & tasks together where equipment would group all its maintenance tasks. I don't think I want this to be just another reminders app.

**Reply by DavidLandes** [reply]
### Problem
I need to change the oil in my car, but I only do this once in a while & forget things like: 
- What oil do I need
- How much oil 
- What filter do I need

### Solution
Home Maintenance can:
- Document the above information
    - Valvoline 5W-30
    - 6 quarts
    - Fram PH10575
- Cost of parts
- Document when the task was last performed
- Remind you of upcoming tasks

**Reply by Pertempto** [reply]
So maybe all the items in the "groups" section should be moved out of this app to another app with a different scope

**Reply by DavidLandes** [reply]
@Pertempto My dilemma is that "Yard" & "Home" don't exactly fit the "equipment" structure. "Yard" doesn't have a model number or manufacturer, but "Equipment" would. Any ideas how we could group these into some common structure?

**Reply by DavidLandes** [reply]
![PXL_20240914_140433972.jpg](https://github.com/user-attachments/assets/ba78c55f-1867-4bd1-94ff-f3608e529cb7)

**Reply by Pertempto** [reply]
Make model number and manufacturer optional fields. Or put them in a different EquipmentInfo table that joins with the group table. Not every Group will have EquipmentInfo, but many will
